kedro/contrib/io/azure/csv_blob.py

```diff
@@ -107,30 +111,60 @@ def __init__(
                 Here you can find all available arguments:
                 https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.to_csv.html
                 All defaults are preserved, but "index", which is set to False.
+            version: If specified, should be an instance of
+                ``kedro.io.core.Version``. If its ``load`` attribute is
+                None, the latest version will be loaded. If its ``save``
+                attribute is None, save version will be autogenerated.
 
         """
-        self._filepath = filepath
         self._container_name = container_name
-        self._credentials = credentials if credentials else {}
-        self._blob_to_text_args = blob_to_text_args if blob_to_text_args else {}
-        self._blob_from_text_args = blob_from_text_args if blob_from_text_args else {}
-        super().__init__(load_args, save_args)
+        self._credentials = copy.deepcopy(credentials) or {}
+        self._blob_to_text_args = copy.deepcopy(blob_to_text_args) or {}
+        self._blob_from_text_args = copy.deepcopy(blob_from_text_args) or {}
+        self._blob_service = BlockBlobService(**self._credentials)
```

Comment:
So we expand `self._creadentials`, so if you do `**{}`, it says invalid syntax, so I think credentials are required and we don't need to do ` or {}` and Optional (sorry this was my earlier suggestion) for `creadentials` arg.

Reply:
No worries, should be back to normal now. :)